### PR TITLE
Spanish NIE (Foreigners ID card) recognizer

### PR DESCRIPTION
**Pull request by areyesfalcon**
## Change Description

Added recognizer: Spanish NIE

## Issue reference

This PR does not fix any issue

## Checklist

- [X] I have reviewed the [contribution guidelines](https://github.com/microsoft/presidio/blob/main/CONTRIBUTING.md)
- [X] I have signed the CLA (if required)
- [X] My code includes unit tests
- [X] All unit tests and lint checks pass locally
- [X] My PR contains documentation updates / additions if required

### CODE REVIEW

**Comment by areyesfalcon** [comment]
@microsoft-github-policy-service agree

**Comment by areyesfalcon** [comment]
Last 2 errors aren't due to my pull request (they aren't directly related with my changes). I don't know how can I resolve these.

File: presidio-analyzer/presidio_analyzer/recognizer_registry.py [hunk 1]
```diff
@@ -236,7 +244,8 @@ def get_recognizers(
         )
 
         if not to_return:
-            raise ValueError("No matching recognizers were found to serve the request.")
+            raise ValueError("No matching recognizers were found to serve the\
```
Comment (RoeyBC) [inline presidio-analyzer/presidio_analyzer/recognizer_registry.py:247]: Looks like this lines fails the e2e tests FYI

Comment (areyesfalcon) [reply presidio-analyzer/presidio_analyzer/recognizer_registry.py]: Oh! Thank you @RoeyBC. I passed flake8 and recommended this change...

I'll try again.

**Comment by areyesfalcon** [comment]
So much errors?

**Review by omri374 — APPROVED** [review]

**Pull request merged**